android/app/src/main/java/com/expensify/chat/AppStateTrackerModule.kt

```diff
@@ -0,0 +1,28 @@
+package com.expensify.chat
+
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+
+import com.expensify.chat.MainActivity
+
+/**
+ * Checks whether the app was relaunched from the app icon in app drawer, not from recent apps.
+ */
+class AppStateTrackerModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(
+    reactContext
+) {
+    override fun getName(): String = "AppStateTracker"
+
+    @ReactMethod
+    fun getWasAppRelaunchedFromIcon(promise: Promise) {
```

Comment:
Could you add a brief comment explaining why this works? I assume MainActivity gets launched, but that's just because of `singleTask` right?

Reply:
I'm not sure if I understand the question 🤔 In `onNewIntent` we check the intent type and if it's a launcher intent we know that the app has been launched from icon, why?
Because launcher intent can be in this method only when the app was in background and icon has been tapped, otherwise there wouldn't be any `onNewIntent` call and app would go through `onCreate -> ...` flow